src/menu.rs

```diff
@@ -1,43 +1,44 @@
+use std::{
+  collections::hash_map::DefaultHasher,
+  hash::{Hash, Hasher},
+};
+
 #[derive(Debug, Clone)]
-pub struct Menu {
+pub struct Menu<'a> {
   pub title: String,
-  pub items: Vec<MenuItem>,
+  pub items: Vec<MenuItem<'a>>,
 }
 
-impl Menu {
-  pub fn new(title: &str, items: Vec<MenuItem>) -> Self {
+impl<'a> Menu<'a> {
+  pub fn new(title: &str, items: Vec<MenuItem<'a>>) -> Self {
     Self {
       title: String::from(title),
       items,
     }
   }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Hash, Copy)]
 /// CustomMenu is a custom menu who emit an event inside the EventLoop.
-pub struct CustomMenu {
-  // todo: replace id by a type that
-  // we can send from the with_menu::<T>()
-  // or we could check to use the UserEvent::<T> but
-  // my latest test failed
-  pub id: String,
-  pub name: String,
+pub struct CustomMenu<'a> {
+  pub _id: MenuId,
+  pub name: &'a str,
```

Comment:
I would prefer `String` at this rate so we don't have to affect so many types with explicit lifetime.

Reply:
yeah the problem is i need the copy trait if we want to use what i’ve done with custom_menu.id() inside the event loop